plugins/aws/cli_provisioner.go

```diff
@@ -25,22 +26,36 @@ func (p awsCLIProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput,
 
 func stripAndReturnProfileFlag(args *[]string) (string, error) {
 	for i, arg := range *args {
+		// if `--profile` is used after `--`, it should not be interpreted as a flag
+		if arg == "--" {
+			break
+		}
+
 		if arg == "--profile" {
 			if i+1 == len(*args) {
 				return "", fmt.Errorf("--profile flag was specified without a value")
 
 			}
 			profile := (*args)[i+1]
 			// Remove --profile flag so aws cli doesn't attempt to assume role by itself
-			*args = removeProfileFlagFromArgs(i, *args)
+			*args = removeArgSequenceFromArgList(i, i+1, *args)
+			return profile, nil
+		} else if strings.Contains(arg, "--profile=") {
+			split := strings.Split(arg, "=")
+			if len(split) != 2 {
+				return "", fmt.Errorf("--profile flag was specified without a value")
+			}
+			profile := split[1]
+			// Remove --profile flag so aws cli doesn't attempt to assume role by itself
+			*args = removeArgSequenceFromArgList(i, i, *args)
 			return profile, nil
 		}
 	}
 	return "", nil
 }
 
-func removeProfileFlagFromArgs(argIndex int, args []string) []string {
-	result := append(args[0:argIndex], args[argIndex+2:]...)
+func removeArgSequenceFromArgList(startIndex, endIndex int, args []string) []string {
+	result := append(args[0:startIndex], args[endIndex+1:]...)
 	return result
 }
```

Comment:
nit: While I appreciate pulling this out into a separate function. Because of the relatively complex arguments, the code might actually become easier to read if we perform the `append()` inline.